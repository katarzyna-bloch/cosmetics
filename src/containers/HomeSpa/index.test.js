import { mount } from "enzyme"
import toJson from "enzyme-to-json"
import { HomeSpa } from "."

jest.mock("../../components/ProductsList", () => "ProductsList")

it("should render correctly", () => {
  const props = {
    fetchProducts: jest.fn(),
    products: [
      { name: "Test", img: "img.jpg", id: "1", price: "10z≈Ç" },
    ],
  }
  const wrapped = mount(<HomeSpa {...props} />)
  expect(toJson(wrapped)).toMatchSnapshot()
  wrapped.update()
  expect(props.fetchProducts).toHaveBeenCalled()
})
