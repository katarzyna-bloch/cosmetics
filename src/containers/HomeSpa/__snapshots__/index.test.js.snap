// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<HomeSpa
  fetchProducts={
    [MockFunction] {
      "calls": Array [
        Array [],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
  products={
    Array [
      Object {
        "id": "1",
        "img": "img.jpg",
        "name": "Test",
        "price": "10zł",
      },
    ]
  }
>
  <styled.div>
    <div
      className="sc-bdfBwQ lbYnYk"
    >
      <styled.h5>
        <h5
          className="sc-gsTCUz AWQJG"
        >
          Profesjonalne SPA w domowym zaciszu, relaks dla ciała i duszy
        </h5>
      </styled.h5>
      <ProductsList
        products={
          Array [
            Object {
              "id": "1",
              "img": "img.jpg",
              "name": "Test",
              "price": "10zł",
            },
          ]
        }
      />
    </div>
  </styled.div>
</HomeSpa>
`;
